### PR TITLE
Introduce Product Manager seat to project governance

### PR DESCRIPTION
**Pull request by azdagron**
This PR amends MAINTAINERS.md to introduce the PM seat that we (@azdagron, @amartinezfayo, @APTy, @mcpherrinm) gained consensus on during the 06/26/20 maintainers call (and @evan2645, separately offline). I'd love a review on the scope and responsibilities to make sure that I've captured them accurately. Please also ensure that they are well aligned with the rest of the project governance and maintenance text.

I think the assignment for the role can be named in the CODEOWNERS file alongside the maintainers.

As we discussed initially, Andres Vega (@anvega) has been actively fulfilling parts of this role for over the last year. His contributions thus far include:

* Curated a community-needs driven theme-based roadmap for SPIFFE/SPIRE.
* Led the certification of the project to attain a Core Infrastructure Initiative Best Practices badge.
* Authored and led the CNCF SIG-Security assessment.
* Drove and represented the project moving from CNCF sandbox level into the incubation stage (which was a huge undertaking!)
* Delivered a number of SPIFFE/SPIRE webinars and trainings.
* Availed himself to the community, facilitating adopters moving SPIRE into production.
* Contributed to general project process organization, and clarification, including established a vulnerability reporting process

Since we discussed this seat, in my view, Andres has continued to demonstrate his ability and willingness to step into this role in a more official capacity. He has good working relationships with the maintainers and many of the community members, and is well positioned to fill the seat. With this consideration, I'd like to nominate Andres as the project's first PM.

If we're all in agreement and after we have finalized and merged this amendment, I can do a follow-on PR that names Andres to this seat in CODEOWNERS.

### CODE REVIEW

File: MAINTAINERS.md [hunk 1]
```diff
@@ -135,6 +135,29 @@ Considering the above, optimism and friendliness should be liberally applied to
...
+
+The role has three primary areas of focus: roadmap management and curation (to ensure the project direction is representative of the community's needs), program management (to help the project deliver on its strategy and meet its intended outcomes), and project management (to align day-to-day activities to meet the SPIRE project requirements).
+
+The product manager must:
+
+* Work with the maintainers to continually ensure a high-quality release of the projects including owning project management, issue triage and identifying all features in the current release cycle
+* Regularly attend maintainer sync calls
+* Participate actively in Request for Comments and feature proposal processes 
+* Track not just features, but ongoing community wide efforts (The word “Effort” is used purposefully. While much work is focused on new or improved features, there is other important ongoing work that isn’t so clearly feature oriented.)
```
Comment (amartinezfayo) [inline MAINTAINERS.md:151]: Perhaps enclosing the last part in parenthesis is not needed?

```suggestion
* Track not just features, but ongoing community wide efforts. The word “efforts” is used purposefully. While much work is focused on new or improved features, there is other important ongoing work that isn’t so clearly feature oriented
```

File: MAINTAINERS.md [hunk 2]
```diff
@@ -135,6 +135,29 @@ Considering the above, optimism and friendliness should be liberally applied to
...
+* Coordinating changes that result from new work across the larger project and provide clarity on the acceptance, prioritization and timeline for all workstreams and efforts
```
Comment (amartinezfayo) [inline MAINTAINERS.md:152]:
```suggestion
* Coordinate changes that result from new work across the larger project and provide clarity on the acceptance, prioritization and timeline for all workstreams and efforts
```

File: MAINTAINERS.md [hunk 3]
```diff
@@ -135,6 +135,29 @@ Considering the above, optimism and friendliness should be liberally applied to
...
+* Communicating major decisions involving release planning to the developer and end user communities through the project media, communication channels, and community events
```
Comment (amartinezfayo) [inline MAINTAINERS.md:153]:
```suggestion
* Communicate major decisions involving release planning to the developer and end user communities through the project media, communication channels, and community events
```

File: MAINTAINERS.md [hunk 4]
```diff
@@ -135,6 +135,29 @@ Considering the above, optimism and friendliness should be liberally applied to
...
+* Managing the relationship between SPIFFE/SPIRE and the CNCF
```
Comment (amartinezfayo) [inline MAINTAINERS.md:154]:
```suggestion
* Manage the relationship between SPIFFE/SPIRE and the CNCF
```

File: MAINTAINERS.md [hunk 5]
```diff
@@ -135,6 +135,29 @@ Considering the above, optimism and friendliness should be liberally applied to
...
+* Support the marketing and promotion of the SPIFFE and SPIRE project through the CNCF with the objective to foster a more secure cloud native ecosystem
+* Coordinate and facilitate discussions on policy review and changes with the TSC
+* Onboarding contributors and welcoming them into the community
```
Comment (amartinezfayo) [inline MAINTAINERS.md:157]:
```suggestion
* Onboard contributors and welcoming them into the community
```

File: MAINTAINERS.md (hunk 1 above)
Comment (azdagron) [reply MAINTAINERS.md:151]: Hmm, maybe we can reword this to:
```suggestion
* Track feature development and ongoing community undertakings
```

**Comment by azdagron** [comment]
Thank you everybody!

**Pull request merged**